generator/integration_tests/tests/golden_kitchen_sink_rest_metadata_decorator_test.cc

```diff
@@ -0,0 +1,307 @@
+             google::test::admin::database::v1::GenerateIdTokenRequest const&) {
+            EXPECT_THAT(context.GetHeader("x-goog-api-client"),
+                        Contains(google::cloud::internal::ApiClientHeader(
+                            "generator")));
+            EXPECT_THAT(context.GetHeader("x-goog-user-project"),
+                        Contains("test-user-project"));
+            EXPECT_THAT(context.GetHeader("x-goog-quota-user"), IsEmpty());
+            EXPECT_THAT(context.GetHeader("x-server-timeout"), IsEmpty());
+            EXPECT_THAT(context.GetHeader("x-goog-request-params"), IsEmpty());
+            return TransientError();
+          });
+
+  internal::OptionsSpan span(
+      Options{}.set<UserProjectOption>("test-user-project"));
+  GoldenKitchenSinkRestMetadata stub(mock);
+  rest_internal::RestContext context;
+  google::test::admin::database::v1::GenerateIdTokenRequest request;
+  auto status = stub.GenerateIdToken(context, request);
+  EXPECT_EQ(TransientError(), status.status());
+}
+
+TEST(KitchenSinkRestMetadataDecoratorTest, WriteLogEntries) {
+  auto mock = std::make_shared<MockGoldenKitchenSinkRestStub>();
+  EXPECT_CALL(*mock, WriteLogEntries)
+      .WillOnce(
+          [](rest_internal::RestContext& context,
+             google::test::admin::database::v1::WriteLogEntriesRequest const&) {
+            EXPECT_THAT(context.GetHeader("x-goog-api-client"),
+                        Contains(google::cloud::internal::ApiClientHeader(
+                            "generator")));
+            EXPECT_THAT(context.GetHeader("x-goog-user-project"), IsEmpty());
+            EXPECT_THAT(context.GetHeader("x-goog-quota-user"),
+                        Contains("test-quota-user"));
+            EXPECT_THAT(context.GetHeader("x-server-timeout"), IsEmpty());
+            EXPECT_THAT(context.GetHeader("x-goog-request-params"), IsEmpty());
+            return TransientError();
+          });
+
+  internal::OptionsSpan span(Options{}.set<QuotaUserOption>("test-quota-user"));
+  GoldenKitchenSinkRestMetadata stub(mock);
+  rest_internal::RestContext context;
+  google::test::admin::database::v1::WriteLogEntriesRequest request;
+  auto status = stub.WriteLogEntries(context, request);
+  EXPECT_EQ(TransientError(), status.status());
+}
+
+TEST(KitchenSinkRestMetadataDecoratorTest, ListLogs) {
+  auto mock = std::make_shared<MockGoldenKitchenSinkRestStub>();
+  EXPECT_CALL(*mock, ListLogs)
+      .WillOnce([](rest_internal::RestContext& context,
+                   google::test::admin::database::v1::ListLogsRequest const&) {
+        EXPECT_THAT(
+            context.GetHeader("x-goog-api-client"),
+            Contains(google::cloud::internal::ApiClientHeader("generator")));
+        EXPECT_THAT(context.GetHeader("x-goog-user-project"), IsEmpty());
+        EXPECT_THAT(context.GetHeader("x-goog-quota-user"), IsEmpty());
+        EXPECT_THAT(context.GetHeader("x-server-timeout"), Contains("3.000"));
```

Comment:
I noticed your sprinkled several different cases for `x-server-timeout` across these tests.  Optional: maybe refactor the formatting to a common function, test that separately, and call the function from the generated code?